src/game/Entities/Player.cpp

```diff
@@ -2062,6 +2064,7 @@ bool Player::ResolvePendingUnmount()
         ResummonPetTemporaryUnSummonedIfAny();
 
     UpdateSpeed(MOVE_RUN, true); // update speed
+    UpdateSpeed(MOVE_FLIGHT, true);
```

Comment:
I've added an unconditional flight speed update on dismount, because I'm unsure whether there can be a pending mount with a ground-mount before the pending unmount is executed, which would override the `m_pendingMountAuraFlying` variable

Reply:
Make it conditional, for two reasons, its a heavy function and previously it was conditional on having the aura as well.